lib/web_ui/dev/safari_installation.dart

```diff
@@ -45,10 +49,75 @@ class SafariArgParser extends BrowserArgParser {
   String get version => _version;
 
   bool _isMobileBrowser;
-
   bool get isMobileBrowser => _isMobileBrowser;
 }
 
+class IosSafariArgParser extends BrowserArgParser {
+  static final IosSafariArgParser _singletonInstance = IosSafariArgParser._();
+
+  /// The [IosSafariArgParser] singleton.
+  static IosSafariArgParser get instance => _singletonInstance;
+
+  String get version => 'iOS ${iosMajorVersion}.${iosMinorVersion}';
+
+  int _pinnedIosMajorVersion;
+  int _iosMajorVersion;
+  int get iosMajorVersion => _iosMajorVersion ?? _pinnedIosMajorVersion;
+
+  int _pinnedIosMinorVersion;
+  int _iosMinorVersion;
+  int get iosMinorVersion => _iosMinorVersion ?? _pinnedIosMinorVersion;
+
+  String _pinnedIosDevice;
+  String _iosDevice;
+  String get iosDevice => _iosDevice ?? _pinnedIosDevice;
+
+  IosSafariArgParser._();
+
+  @override
+  void populateOptions(ArgParser argParser) {
+    final YamlMap browserLock = BrowserLock.instance.configuration;
+    _pinnedIosMajorVersion = browserLock['ios-safari']['majorVersion'] as int;
+    _pinnedIosMinorVersion = browserLock['ios-safari']['minorVersion'] as int;
+    _pinnedIosDevice = browserLock['ios-safari']['device'] as String;
+    argParser
+      ..addOption('majorVersion',
```

Comment:
Do you plan to use the same command for Android emulator? If so, how will the command-line syntax structure differentiate between Android and iOS. Do you plan to use the value of `--device` to determine what other options mean?

Reply:
done. added a type option.